lib/Doctrine/MongoDB/GridFS.php

```diff
@@ -288,4 +293,24 @@ protected function doUpdate(array $query, array $newObj, array $options = [])
         $options = isset($options['safe']) ? $this->convertWriteConcern($options) : $options;
         return $this->mongoCollection->update($query, $newObj, $options);
     }
+
+    /**
+     * Executes a closure with a temporary primary read preference.
+     *
+     * @param \Closure $closure
+     *
+     * @return mixed
+     */
+    private function withPrimaryReadPreference(\Closure $closure)
+    {
```

Comment:
I left it out because I didn't think the overhead of changing the read preference to `primary` and setting it back would be bad. The alternative in this case would be to compare the original read preference and conditionally change it to master. I can change it if you think it will improve performance.

Reply:
Makes sense as-is. This keeps things simpler.